src/utils/temp_folder.ts

```diff
@@ -2,11 +2,61 @@ import * as fs from 'fs';
 import * as os from 'os';
 import path from 'path';
 
+const isWindows = os.platform() === 'win32';
+
+function getOrCreateTempBaseFolder(): string {
+	if (isWindows) {
+		return getValidWindowsTempPath() ?? getManualWindowsTempPath();
+	} else {
+		return getValidUnixTempPath() ?? getManualUnixTempPath();
+	}
+}
+
+function getValidWindowsTempPath(): string | null {
+	const envTempFolder = process.env['TEMP'] ?? process.env['TMP'];
+	if (envTempFolder) {
+		if (fs.existsSync(envTempFolder)) {
+			return envTempFolder;
+		} else {
+			const userProfile = process.env['USERPROFILE'];
+			if (userProfile) {
+				const tempPath = path.join(userProfile, 'AppData', 'Local', 'Temp');
+				if (fs.existsSync(tempPath)) {
+					return tempPath;
+				}
+			}
+		}
+	}
+	return null;
+}
+
+function getValidUnixTempPath(): string | null {
+	const envTempFolder = process.env['TMPDIR'] ?? process.env['TMP'] ?? process.env['TEMP'];
+	if (envTempFolder) {
+		if (fs.existsSync(envTempFolder)) {
+			return envTempFolder;
+		} else {
+			const tempPath = path.join('tmp');
```

Comment:
`tempPath` should be `/tmp` directly or `path.join('/tmp')`, but there's no reason to use `path.join` here.